src/pages/Home.jsx

```diff
@@ -5,126 +5,126 @@ import DocApi from "@/util/document_apis";
 import { Link } from "react-router-dom";
 
-// import DocSortingExpandIcon from "@/assets/home_doc_sorting_expand_icon.svg";
-// import Typography from '@mui/material/Typography';
-// import Popover from '@mui/material/Popover';
-// import { useState } from "react";
-
-
-// const SortingDropDownMenu = (props) => {
-//   return <div className={'flex flex-col bg-red-100 w-[100px] border-solid rounded ' + props.className}>
-//     <div>1</div>
-//     <div>2</div>
-//     <div>3</div>
-//   </div>
-// }
-
-// const DocumentsSelectionTopBar = () => {
-//   const [anchorEl, setAnchorEl] = useState(null);
-
-//   const handlePopoverOpen = (event) => {
-//     setAnchorEl(event.currentTarget);
-//   };
-
-//   const handlePopoverClose = () => {
-//     setAnchorEl(null);
-//   };
-
-//   const open = Boolean(anchorEl);
-
-//   return <div className='pt-5'>
-//     <div className='text-xl font-bold'>
-//       <div className="flex flex-row"> 
-//         <div>Documents</div>
-//         <img src={DocSortingIcon} className="ml-5"/>
-//         <Typography
-//           aria-owns={open ? 'mouse-over-popover' : undefined}
-//           aria-haspopup="true"
-//           onMouseEnter={handlePopoverOpen}
-//           // onMouseLeave={handlePopoverClose}
-//         >
-//           <img src={DocSortingExpandIcon}/>
-//         </Typography>
-//       </div>
-//       <Popover
-//         id="mouse-over-popover"
-//         sx={{
-//           pointerEvents: 'none',
-//         }}
-//         open={open}
-//         anchorEl={anchorEl}
-//         anchorOrigin={{
-//           vertical: 'bottom',
-//           horizontal: 'left',
-//         }}
-//         transformOrigin={{
-//           vertical: 'top',
-//           horizontal: 'left',
-//         }}
-//         onClose={handlePopoverClose}
-//         onMouseLeave={handlePopoverClose}
-//         disableRestoreFocus
-//       >
-//         <SortingDropDownMenu/>
-//       </Popover>
-//     </div>
-//   </div>
-// }
-
-const DocumentsSelectionTopBar = () => {
+import DocSortingExpandedIcon from "@/assets/home_doc_sorting_expand_icon.svg";
+import DocSortingExpandIcon from "@/assets/expand_right.svg"
+import Popover from '@mui/material/Popover';
+import { useReducer, useState } from "react";
+import Clickable from "@/components/common/Clickable";
+import { SORTING_OPTIONS } from "../util/constants";
+
+
+const DocumentsSelectionTopBar = ({ resort }) => {
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [selectedSortKey, setselectedSortKey] = useState(SORTING_OPTIONS.name)
+
+  const SortingDropDownMenu = ({ selectedSortKey, setselectedSortKey }) => {
+    const sortDisplay = [
+      { name: "Name", key: SORTING_OPTIONS.name },
+      { name: "Date Uploaded", key: SORTING_OPTIONS.date },
+      { name: "File Type", key: SORTING_OPTIONS.file }
+    ]
+    return <div className={'flex flex-col border-solid rounded items-start p-2 divide-y divide-black'}>
+      {sortDisplay.map(so => {
```

Comment:
What’s this “so” thing? Usually I see it in the form “((var) => {…})”

Reply:
oh yeah it's a js thing where if you only have one param you don't need the brackets, it's a function like normal